editor/filesystem_dock.cpp

```diff
@@ -426,22 +426,34 @@ void FileSystemDock::set_display_mode(DisplayMode p_display_mode) {
 void FileSystemDock::_update_display_mode(bool p_force) {
 	// Compute the new display mode.
 	if (p_force || old_display_mode != display_mode) {
-		button_toggle_display_mode->set_pressed(display_mode == DISPLAY_MODE_SPLIT);
 		switch (display_mode) {
 			case DISPLAY_MODE_TREE_ONLY:
+				button_toggle_display_mode->set_icon(get_theme_icon(SNAME("Panels1"), SNAME("EditorIcons")));
```

Comment:
```suggestion
				button_toggle_display_mode->set_icon(get_editor_theme_icon(SNAME("Panels1")));
```
This was just merged, so you should use the new method.